pkg/recipes/terraform/config/config.go

```diff
@@ -189,8 +284,9 @@ func getProviderConfigs(ctx context.Context, requiredProviders []string, ucpConf
 		if err != nil {
 			return nil, err
 		}
+
 		if len(config) > 0 {
-			providerConfigs[provider] = config
+			providerConfigs[provider] = []map[string]any{config}
```

Comment:
What is causing this to change? Return type is `map[string]any`

Reply:
We're making sure the structure of provider config is the same when UCP provider config is built or we get it from env-recipe config. the UCP provider config was earlier generating it in map[string]any format

Reply:
We chatted offline and decided to the config provider map value strongly typed to avoid assumptions and better readability/maintainability of this code path.

Reply:
updated in latest commit + rebased. Pls take a look.